src/main/java/com/ubiquitech/leaveTrack/dao/EmployeeDaoImpl.java

```diff
@@ -30,46 +26,32 @@ public void createEmployee(Employee employee) {
 
     @Override
     public Employee getEmployeeById(Long id) {
-        Session session = sessionFactory.openSession();
-        Employee employee=(Employee) session.get(Employee.class,id);
-        session.close();
-        return employee;
+        Session session = sessionFactory.getCurrentSession();
+        return (Employee) session.get(Employee.class,id);
     }
 
     @Override
     public List<Object[]> getEmployeeNames() {
-      List<Object[]> employees=null ;
-        Session session = sessionFactory.openSession();
-        Criteria criteria=session.createCriteria(Employee.class);
```

Comment:
The hibernate criteria API is unfortunately a bit messy and makes code harder to understand. We avoid using it. For constructing really complex select statements we use another similar but better project called QueryDSL. But in general, try stick to HQL - Hibernate Query Language, which is similar in many ways to SQL. You'll see its easier to understand